### ISSUE TITLE
undefined symbol: gsl_rng_uniform_int

### ISSUE THREAD

**Issue by AdrianSosic** [slinderman/pypolyagamma#29]
Hi Scott,

Thanks for sharing your code. Unfortunately, I am having problems executing it on my Linux machine. While everything works fine on macOS, I get the following error on Linux:

> Traceback (letzter Anruf zuletzt):
>   Datei "<string>", Zeile 1, in <module>>.
>   Datei "/home/asosic/Checkouts/p3/pypolyagamma/pypolyagamma/__init__.py", Zeile 3, in <modul>
>     aus pypolyagamma.pypolyagamma Import PyRNG, PyPolyaGamma                            
> ImportError: /home/asosic/Checkouts/p3/pypolyagamma/pypolyagamma/pypolyagamma.cpython-36m-x86_64-linux-gnu.so: undefiniertes Symbol: gsl_rng_uniform_int  

            

The minimal code to reproduce the error is:
```
virtualenv -p python3 venv
source venv/bin/activate
pip install cython    
git clone https://github.com/slinderman/pypolyagamma
cd pypolyagamma
pip install -e .
cd ..
python -c "import pypolyagamma"
```

As I said before, the above runs perfectly fine on macOS. Do you have any clue what is going on here?

Best regards,
Adrian

**Comment by slinderman** [comment]
Hi Adrian,

Looks like it's not linking properly.  Can you try rerunning the above but with `pip install -v - e .` and post the output?  You should see a bunch of calls like `gcc ... -c deps/gsl/...` followed by a call to `g++`.   I want to see if those look as expected.

I was not able to repro this on my linux machine, and the Travis builds are working properly, also on linux.  I'm not sure what could be different about your setup.

Best,
Scott

**Comment by slinderman** [comment]
Thanks for sharing.  I don't see anything particularly strange in the logs...  no errors in compilation or linking reported. 

Why don't you try this instead.  First install GSL manually, e.g. on Ubuntu:
```
sudo apt-get install libgsl-dev
```

Then install pypolyagamma by linking against the system-wide GSL library (rather than downloading GSL and compiling necessary source files, as the master branch does).  You can do this with the `condagsl` branch.  These steps should do it:

```
virtualenv -p python3 venv
source venv/bin/activate
pip install cython    
git clone https://github.com/slinderman/pypolyagamma
cd pypolyagamma
git checkout condagsl
pip install -v -e .
cd ..
python pypolyagamma/test/test_basics.py
```

**Comment by AdrianSosic** [comment]
Hi Scott,

good news: it's running -- yet, I still don't know what the problem was. It turned out that everything works fine if I run the code in a conda environment. Apparently, the linking problem does not occur there. However, I used had to use the master branch (or, alternatively, simply "pip install pypolygamma") since the proposed condagsl approach didn't work either. 

By the way, while testing, I also noticed that the pip installation only works with Python 3.6 but not with 3.7. The error looks pretty much like the one described here:
https://github.com/rkern/line_profiler/issues/132](url)
which seems to be related to some precompiled files.

Anyway, thanks a lot for your help!
